managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -2988,57 +2991,88 @@ public void testManagedLedgerWithReadEntryTimeOut() throws Exception {
 
```

Comment:
Does the original test case fail with your change? If not, I suggest you add a new case instead of modifying an existing case. Otherwise, it looks like a breaking change.